### PR TITLE
Finish task 6 with test suite, added model to check for id

### PR DESCRIPTION
**Pull request by valentinbiz**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -124,5 +124,53 @@ describe("API testing", () => {
...
+                  body: expect.any(String),
+                  votes: expect.any(Number),
+                  comment_id: expect.any(Number),
+                  created_at: expect.any(String),
+                })
+              );
+            });
+          });
+      });
+      test("200: Should return a 200 status with no body returned when the article exists but there are no comments", () => {
```
Comment (robert112233) [inline __tests__/app.test.js:148]: Great test to differentiate 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -124,5 +124,53 @@ describe("API testing", () => {
...
+        return request(app)
+          .get("/api/articles/1/comments")
+          .expect(200)
+          .then(({ body: { comments } }) => {
+            expect(comments).toBeInstanceOf(Array);
+            expect(comments).toHaveLength(11);
+            expect(comments).toBeSortedBy("created_at", { descending: true });
+          });
+      });
+      test("404: It should return an error when the id is valid but non-existent", () => {
```
Comment (robert112233) [inline __tests__/app.test.js:166]: Great sad path test here, what happens if I make a request to /api/articles/four/comments ? what status code would that be?

File: controllers/controllers.articles.js [hunk 3]
```diff
@@ -16,7 +18,20 @@ const getArticleById = (request, response, next) => {
     .catch((error) => next(error));
 };
 
+const getCommentsByArticleId = (request, response, next) => {
+  const articleId = request.params.article_id;
+  const promises = [selectCommentsByArticleId(articleId)];
+  promises.push(checkIfIdExists(articleId));
```
Comment (robert112233) [inline controllers/controllers.articles.js:24]: Nice incorporation of yesterdays lecture 👍

**Review by robert112233 — CHANGES_REQUESTED** [review]
Great work, just missing one test 👍 Review changes then submit another PR for review :)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -171,6 +171,14 @@ describe("API testing", () => {
             expect(msg).toBe("Not found!");
           });
       });
+      test.only("400: It should return an error when the id provided is of invalid type", () => {
```
Comment (robert112233) [inline __tests__/app.test.js]: test.only !

**Review by robert112233 — CHANGES_REQUESTED** [review]
Remove test.only

File: __tests__/app.test.js (hunk 4 above)
Comment (valentinbiz) [reply __tests__/app.test.js]: Fixed that in the latest commit

**Review by robert112233 — APPROVED** [review]

**Pull request merged**